File: samples_c/s65_ctype_isalpha_highbyte.c
#include <stdio.h>
#include <ctype.h>

/* Girdi: tek satir metin.
   Cikti: "<alpha_sayisi> <digit_sayisi>".
   NOT: Kok Neden C (char isaretliligi) icin 4. bagimsiz ornek. isalpha()/
   isdigit() PLAIN `char` ile cagrilir (unsigned char'a donusturulmeden) -
   standartta bu, EOF disindaki negatif degerler icin tanimsizdir; bu
   derleme/kutuphane ortaminda (glibc/UCRT benzeri tablo tabanli ctype)
   deterministik ama platforma bagli sonuc uretir - kok nedeni C'nin diger
   bir yuzunu (kutuphane API sozlesmesindeki isaretlilik varsayimi) sinar. */
int main(void) {
    char buf[4096];
    if (fgets(buf, sizeof(buf), stdin) == NULL) return 0;
    int alpha = 0, digit = 0;
    for (int i = 0; buf[i] != '\0'; i++) {
        if (buf[i] == '\n' || buf[i] == '\r') break;
        char c = buf[i];
        if (isalpha(c)) alpha++;
        if (isdigit(c)) digit++;
    }
    printf("%d %d\n", alpha, digit);
    return 0;
}

File: translations_rust__gemini/s65_ctype_isalpha_highbyte.rs
use std::io::{self, Read};

fn main() {
    let mut buf = [0u8; 4096];
    let mut len = 0;
    let mut stdin = io::stdin().lock();

    while len < 4095 {
        let mut b = [0u8; 1];
        match stdin.read(&mut b) {
            Ok(1) => {
                buf[len] = b[0];
                len += 1;
                if b[0] == b'\n' {
                    break;
                }
            }
            _ => break,
        }
    }

    if len == 0 {
        return;
    }

    let mut alpha = 0;
    let mut digit = 0;

    for &b in &buf[..len] {
        if b == b'\0' || b == b'\n' || b == b'\r' {
            break;
        }
        if b.is_ascii_alphabetic() {
            alpha += 1;
        }
        if b.is_ascii_digit() {
            digit += 1;
        }
    }

    println!("{} {}", alpha, digit);
}
